tests/applications/test_junebug_message_api.py

```diff
@@ -0,0 +1,667 @@
+        content=content,
+    )
+
+
+def mkev(message_id: str, event_type: EventType, **fields) -> Event:
+    if event_type == EventType.ACK:
+        fields.setdefault("sent_message_id", message_id)
+    return Event(
+        user_message_id=message_id,
+        event_type=event_type,
+        **fields,
+    )
+
+
+async def test_inbound_message_amqp(jma_worker, jma_ro, http_server):
+    """
+    Inbound messages are forwarded to the configured URL.
+
+    This test sends the inbound message over AMQP rather than calling
+    the handler directly.
+    """
+    msg = mkmsg("hello")
+
+    with fail_after(2):
+        await jma_ro.publish_inbound(msg)
+        req = await http_server.receive_req()
+        await http_server.send_rsp(RspInfo())
+
+    assert req.path == "message"
+    assert req.headers["Content-Type"] == "application/json"
+    assert req.body_json["content"] == "hello"
+    assert req.body_json["to"] == "123"
+
+
+async def test_inbound_message(jma_worker, http_server):
+    """
+    Inbound messages are forwarded to the configured URL.
+
+    This test calls the handler directly so we know when it's finished.
+    """
+    msg = mkmsg("hello")
+
+    with fail_after(2):
+        async with handle_inbound(jma_worker, msg):
+            req = await http_server.receive_req()
+            await http_server.send_rsp(RspInfo())
+
+    assert req.body_json["content"] == "hello"
+    assert req.body_json["to"] == "123"
+
+
+async def test_send_message_bad_response(jma_worker, http_server, caplog):
+    """
+    If an inbound message results in an HTTP error, the error and
+    message are logged.
+    """
+    msg = mkmsg("hello")
+
+    with fail_after(2):
+        async with handle_inbound(jma_worker, msg):
+            await http_server.receive_req()
+            await http_server.send_rsp(RspInfo(code=500))
+
+    [err] = [log for log in caplog.records if log.levelno >= logging.ERROR]
+    assert "Error sending message, received HTTP code 500" in err.getMessage()
+
+
+async def test_send_message_basic_auth_url(worker_factory, http_server):
+    """
+    If mo_message_url has credentials in it, those get sent as an
+    Authorization header.
+    """
+    url = f"{http_server.bind}/message".replace("http://", "http://foo:bar@")
+    config = mk_config(http_server, mo_message_url=url)
+
+    msg = mkmsg("hello")
+
+    async with worker_factory.with_cleanup(JunebugMessageApi, config) as jma_worker:
+        await jma_worker.setup()
+        with fail_after(2):
+            async with handle_inbound(jma_worker, msg):
+                req = await http_server.receive_req()
+                await http_server.send_rsp(RspInfo())
+
+    basic = b64encode(b"foo:bar").decode()  # base64 is all bytes, not strs.
+    assert req.headers["Authorization"] == f"Basic {basic}"
+
+
+async def test_send_message_auth_token(worker_factory, http_server):
+    """
+    If mo_message_url has credentials in it, those get sent as an
```

Comment:
Isn't this set in the `mo_message_url_auth_token ` config field rather than the `mo_message_url`?

Reply:
Whoops, forgot to update the docstring.